src/test/java/com/clouway/chita/HttpClientTest.java

```diff
@@ -378,17 +384,20 @@ public void cacheReadingOfResponse() throws Exception {
   }
 
   @Test
-  public void readingRequestTimeout() throws Exception {
+  public void readingRequestTimeout() {
     HttpRequest post = httpRequest(new TargetUrl(serverUrl, serviceUrl))
             .post("")
             .readTimeout(1)
             .build();
 
     server.sleepFor(2000);
-    HttpResponse response = httpClient.execute(post);
-
-    assertThat(response.code(), is(408));
-    assertThat(response.isTimedOut(), is(true));
+    HttpResponse response = null;
+    try {
+      response = httpClient.execute(post);
+      fail("IOException was not thrown when connection timeouts ? ");
+    } catch (IOException e) {
+      assertThat(response, is(nullValue()));
```

Comment:
btw, as you know SocketTimeoutException is IOException, so something like is valid: 

```java 
  try {
    response = client.execute(request);
  } catch(ConnectionTimeoutException e) {
     // mark device as something
  } catch(IOException e) {
     // generic error has occurred and server is not responding 
  }
```